internal/clients/http/azure/create_vm.go

```diff
@@ -0,0 +1,395 @@
+		logger.Error().Err(err).Msgf("cannot create public IP address")
+		return nil, err
+	}
+	logger.Trace().Msgf("Using public IP address id=%s", *publicIP.ID)
+
+	// network security group
+	nsg, err := c.createNetworkSecurityGroup(ctx, location, resourceGroupName, nsgName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create network security group")
+		logger.Error().Err(err).Msgf("cannot create network security group")
+		return nil, err
+	}
+	logger.Trace().Msgf("Using network security group id=%s", *nsg.ID)
+
+	networkInterface, err := c.createNetworkInterface(ctx, location, resourceGroupName, subnet, publicIP, nsg, nicName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create network interface")
+		logger.Error().Err(err).Msgf("cannot create network interface")
+		return nil, err
+	}
+	logger.Trace().Msgf("Using network interface id=%s", *networkInterface.ID)
+
+	virtualMachine, err := c.createVirtualMachine(ctx, location, resourceGroupName, networkInterface, imageID, sshKeyBody, vmName, diskName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create virtual machine")
+		logger.Error().Err(err).Msgf("cannot create virtual machine")
+		return nil, err
+	}
+	logger.Debug().Msgf("Created virtual machine id=%s", *virtualMachine.ID)
+
+	return virtualMachine, nil
+}
+
+func (c *client) createResourceGroup(ctx context.Context, name string, location string) (*armresources.ResourceGroup, error) {
+	resourceGroupClient, err := c.newResourceGroupsClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armresources.ResourceGroup{
+		Location: ptr.To(location),
+		Tags:     map[string]*string{"sample-rs-tag": ptr.To("sample-tag")}, // resource group update tags
+	}
+
+	resp, err := resourceGroupClient.CreateOrUpdate(ctx, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp.ResourceGroup, nil
+}
+
+func (c *client) createVirtualNetwork(ctx context.Context, resourceGroupName string, name string) (*armnetwork.VirtualNetwork, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "createVirtualNetwork")
+	defer span.End()
+
+	vnetClient, err := c.newVirtualNetworksClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armnetwork.VirtualNetwork{
+		Location: to.Ptr(location),
+		Properties: &armnetwork.VirtualNetworkPropertiesFormat{
+			AddressSpace: &armnetwork.AddressSpace{
+				AddressPrefixes: []*string{
+					to.Ptr("10.42.0.0/16"),
+				},
+			},
+			//Subnets: []*armnetwork.Subnet{
+			//	{
+			//		Name: to.Ptr(subnetName+"3"),
+			//		Properties: &armnetwork.SubnetPropertiesFormat{
+			//			AddressPrefix: to.Ptr("10.1.0.0/24"),
+			//		},
+			//	},
+			//},
+		},
+	}
+
+	pollerResponse, err := vnetClient.BeginCreateOrUpdate(ctx, resourceGroupName, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Yeah this will be super slow, Azure is way slower than EC2. We need to move this into a job ASAP.

I am fine merging this as-is tho.

Reply:
I'll move it to job in a second commit, it should not be as hard, I just wanted to have a way of testing the client, but I'll move the changes to the service to separate commit :)